Sources/NIO/Bootstrap.swift

```diff
@@ -966,4 +1133,134 @@ public final class NIOPipeBootstrap {
+        }
+        
+        fileprivate enum ShorthandChannelOption {
+            case disableAutoRead
+            case allowRemoteHalfClosure(Bool)
+            
+            func applyOption(to bootstrap: NIOPipeBootstrap) {
+                switch self {
+                case .disableAutoRead:
+                    _ = bootstrap.channelOption(ChannelOptions.autoRead, value: false)
+                case .allowRemoteHalfClosure(let value):
+                    _ = bootstrap.channelOption(ChannelOptions.allowRemoteHalfClosure, value: value)
+                }
+            }
+        }
+    }
+}
+
+// Hashable for the convenience of users.
+extension NIOPipeBootstrap.Option: Hashable {}
+extension NIOPipeBootstrap.Option.ShorthandChannelOption: Hashable {}
+
+/// Approved shorthand datagram channel options.
+extension NIOPipeBootstrap.Option {
+    /// Option to disable autoRead
+    /// - See: ChannelOptions.autoRead
+    public static let disableAutoRead = NIOPipeBootstrap.Option(.disableAutoRead)
+    
+    /// - See: `AllowRemoteHalfClosureOption`.
+    public static let allowRemoteHalfClosure =
+        NIOPipeBootstrap.Option(.allowRemoteHalfClosure(true))
+    
+    /// - See: `AllowRemoteHalfClosureOption`.
+    public static func allowRemoteHalfClosure(_ value: Bool) ->
+        NIOPipeBootstrap.Option {
+        return NIOPipeBootstrap.Option(.allowRemoteHalfClosure(value))
+    }
+}
+
+
+
+// ------------------------
+
+public protocol NIOTCPOptionAppliable {
+    func applyOption<Option: ChannelOption>(_ option: Option, value: Option.Value) -> Self
+}
+
+extension ServerBootstrap : NIOTCPOptionAppliable {
+    public func applyOption<Option>(_ option: Option, value: Option.Value) -> Self where Option : ChannelOption {
+        return self.childChannelOption(option, value: value)
+    }
+}
+
+extension ClientBootstrap : NIOTCPOptionAppliable {
+    public func applyOption<Option>(_ option: Option, value: Option.Value) -> Self where Option : ChannelOption {
+        return self.channelOption(option, value: value)
+    }
+}
+
+/// A channel option which can be applied to a bootstrap or similar using shorthand notation.
+/// - See: ClientBootstrap.channelOptions(_ options: [Option])
+public struct NIOTCPShorthandOption  {
+    private let data: ShorthandOption
+    
+    private init(_ data: ShorthandOption) {
+        self.data = data
+    }
+    
+    /// Apply the contained option to the supplied object (almost certainly bootstrap) using the default mapping.
+    /// - Parameter to: object to apply this option to.
+    /// - Returns: the modified object
+    public func applyOption<T : NIOTCPOptionAppliable>(with: T) -> T {
+        return data.applyOption(with: with)
+    }
+    
+    fileprivate enum ShorthandOption {
+        case reuseAddr
+        case disableAutoRead
+        case allowRemoteHalfClosure(Bool)
+        
+        func applyOption<T : NIOTCPOptionAppliable>(with: T) -> T {
```

Comment:
This is probably a good place to use Swift's distinction between an argument label and the variable name. The function code uses `with.applyOption`, which reads very strangely. If this becomes `(to bootstrap: T)` then the code can use `bootstrap.applyOption`, which reads much more clearly.

Reply:
I have done something similar to suggestion with new implementation